### PR TITLE
DAOS-11423 ec: enable EC parity rotation

### PR DESCRIPTION
**Pull request by wangdi1**
enable EC parity rotation.

Signed-off-by: Di Wang <di.wang@intel.com>

### CODE REVIEW

**Review by johannlombardi — APPROVED** [review]

**Review by johannlombardi — DISMISSED** [review]

**Review by liuxuezhao — APPROVED** [review]

**Comment by wangdi1** [comment]
seems unrelated failure, just re-triggered the test.

**Comment by wangdi1** [comment]
sorry, have to rebase to fix another issue.  by [ ](https://github.com/daos-stack/daos/pull/10267)

**Comment by johannlombardi** [comment]
Still failing?

**Comment by wangdi1** [comment]
> Still failing?

These failure is not related with EC at all. let me check

**Comment by wangdi1** [comment]
The failure are unrelated. I pushed PR-10279 and PR-10097 to fix them, and rebase this one. Thanks.

**Comment by wangdi1** [comment]
Passed all tests this time. please check

**Review by liuxuezhao — APPROVED** [review]

File: src/object/cli_obj.c [hunk 1]
```diff
@@ -1421,7 +1421,7 @@ dc_obj_open(tse_task_t *task)
 	if (rc)
 		D_GOTO(fail_layout_created, rc);
 
-	obj->cob_ec_parity_rotate = 0;
```
Comment (Nasf-Fan) [inline src/object/cli_obj.c:1424]: Why not define "ec_parity_rotate", then set "obj->cob_ec_parity_rotate = ec_parity_rotate" as test logic does. That will be more convenient for enable/disable the EC rotation.

**Review by Nasf-Fan — APPROVED** [review]

File: src/object/cli_obj.c (hunk 1 above)
Comment (wangdi1) [reply src/object/cli_obj.c:1424]: oh, this is temporary solution, I will remove all of these parity_rotate flags, once I added the layout version.

**Pull request merged**